test/python/test_localjob.py

```diff
@@ -15,188 +15,154 @@
-        # contingency table
-        ctable = numpy.array([[counts_qx.get(key, 0) for key in states],
-                              [counts_ex.get(key, 0) for key in states]])
-        contingency = chi2_contingency(ctable)
-        self.log.info('chi2_contingency: %s', str(contingency))
-        self.assertGreater(contingency[1], 0.01)
-
-    @slow_test
-    def test_run_async(self):
-        if sys.platform == 'darwin':
-            LocalJob._executor = futures.ThreadPoolExecutor(max_workers=2)
-        else:
-            LocalJob._executor = futures.ProcessPoolExecutor(max_workers=2)
-        try:
-            backend = self._provider.get_backend('local_qasm_simulator_cpp')
-        except KeyError:
-            backend = self._provider.get_backend('local_qasm_simulator_py')
-        num_qubits = 15
-        qr = QuantumRegister(num_qubits, 'q')
-        cr = ClassicalRegister(num_qubits, 'c')
-        qc = QuantumCircuit(qr, cr)
-        for i in range(num_qubits-1):
-            qc.cx(qr[i], qr[i+1])
-        qc.measure(qr, cr)
-        qobj = qiskit._compiler.compile(qc, backend)
-        quantum_job = QuantumJob(qobj, backend, preformatted=True)
-        num_jobs = 5
-        job_array = [backend.run(quantum_job) for _ in range(num_jobs)]
-        found_async_jobs = False
-        timeout = 30
-        start_time = time.time()
-        self.log.info('testing with simulator: %s', backend.name)
-        while not found_async_jobs:
-            check = sum([job.running for job in job_array])
-            if check >= 2:
-                self.log.info('found %d simultaneous jobs', check)
-                found_async_jobs = True
-            if all([job.done for job in job_array]):
-                self.log.warning('all jobs completed before simultaneous jobs '
-                                 'could be detected')
-                break
-            for job in job_array:
-                self.log.info('%s %s %s', job.status['status'],
-                              job.running, check)
-            self.log.info('%s %.4f', '-'*20, time.time()-start_time)
-            if time.time() - start_time > timeout:
-                raise TimeoutError('failed to see multiple running jobs after '
-                                   '{0} s'.format(timeout))
-            time.sleep(1)
-
-        # Wait for all the jobs to finish.
-        # TODO: this causes the test to wait until the 15 qubit jobs are
-        # finished, which might take long (hence the @slow_test). Waiting for
-        # the result is needed as otherwise the jobs would still be running
-        # once the test is completed, causing failures in subsequent tests as
-        # the executor's queue might be overloaded.
-
-        _ = [job.result() for job in job_array]
+        with mocked_simulator_binaries(),\
+             patch.object(LocalJob, '__init__', return_value=None,
+                          autospec=True):
+
+            for backendConstructor in self._backends:
```

Comment:
Can you use [`with self.subTest()`](https://docs.python.org/3/library/unittest.html#subtests) here? Mostly for allowing better handling of error in case the test fails, as it allows each assertion inside the block to be tested independently.